bin-src/tasks/upload.js

```diff
@@ -166,6 +175,51 @@ export const uploadStorybook = async (ctx, task) => {
 
   ctx.uploadedBytes = total;
   ctx.isolatorUrl = new URL('/iframe.html', domain).toString();
+}
+
+async function uploadAsZipFile(ctx, task) {
+  const zipped = await makeZipFile(ctx);
+  const { path, size: total } = zipped;
+  const { getZipUploadUrl } = await ctx.client.runQuery(TesterGetZipUploadUrlMutation);
+  const { domain, url, sentinelUrl } = getZipUploadUrl;
+
+  task.output = starting(ctx).output;
+
+  try {
+    await uploadZip(ctx, path, url, total, (progress) => {
+      if (ctx.options.interactive) {
+        const percentage = Math.round((progress / total) * 100);
+        task.output = uploading({ percentage }).output;
+      }
+    });
+  } catch (e) {
+    if (path === e.message) {
+      throw new Error(failed({ path }).output);
+    }
+    throw e;
+  }
+
+  ctx.uploadedBytes = total;
+  ctx.isolatorUrl = new URL('/iframe.html', domain).toString();
+
+  return waitForUnpack(ctx, sentinelUrl);
+}
+
+export const uploadStorybook = async (ctx, task) => {
+  if (ctx.skip) return;
+  transitionTo(preparing)(ctx, task);
+
+  if (ctx.options.zip) {
+    try {
+      await uploadAsZipFile(ctx, task);
+    } catch (err) {
+      ctx.log.debug('error uploading zip file', err);
```

Comment:
Our usual pattern for logs is to have an object with `err` property (and potentially other stuff) first, followed by an optional string message. @tmeasday can probably explain why.

```suggestion
      ctx.log.debug({ err }, 'Error uploading zip file');
```

 Also, sentences start with a capital letter. But that's just my OCD.